File: Last Class - 第一題-循序檔案/Teacher.cpp
#include "Teacher.h"
#include <string>
using namespace std;
Teacher::Teacher()
{

}
Teacher::Teacher(string _n, int _a, double _s)
{
	name = _n;
	age = _a;
	score = _s;
}
void Teacher::set(string _n, int _a, double _s)
{
	name = _n;
	age = _a;
	score = _s;
}
void Teacher::setName(string _n)
{
	name = _n;
}
void Teacher::setAge(int _a)
{
	age = _a;
}
void Teacher::setScore(double _s)
{
	score = _s;
}
string Teacher::getName()
{
	return name;
}
int Teacher::getAge()
{
	return age;
}
double Teacher::setScore()
{
	return score;
}

File: Last Class - 第五題-Python 物件組合/Teacher.py
from Body import Body


class Teacher(Body):
    def __init__(self,name,body,salary):
        self.__name = name
        self.__body = body
        self.__salary = salary

    def setName(self,name):
        self.__name = name

    def getName(self):
        return self.__name

    def setBody(self,body):
        self.__body = body

    def getBody(self):
        return self.__body

    def setSalary(self,salary):
        self.__salary = salary

    def getSalary(self):
        return self.__salary

    def toString(self):
        return "{} {} {}".format(self.__name,self.__body.toString(),self.__salary)